feat: ES 7.x client support

jywadhwani opened the pull request:
This involves ES client side changes to support ES 7.9.3 (we are currently at version 5.6.8). A separate module is created which makes use of ES 7.9.3 client.

## Checklist

- [X] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [X] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

jplaisted commented:
Were there any code changes required? Or were the only changes required build files?

jywadhwani commented:
> Were there any code changes required? Or were the only changes required build files?

There were code changes as well.

jplaisted commented:
Can you describe them in commit / PR description? would make this review easier.

jywadhwani commented:
> Can you describe them in commit / PR description? would make this review easier.

Added. Let me know if there is a better way to describe the code changes.

jplaisted commented:
I think this is fine for now.

Thinking about it, a "better" way, if we do something similar in the future, would be to have a PR with two commits. The first commit just verbatim copies the code. The second commit modifies it. Then I can view the diff between the two commits. Not worth messing around with for now, but we can consider that in the future.

jplaisted commented on dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/search/ESSearchDAOTest.java:
```diff
@@ -0,0 +1,230 @@
+
+  private static String loadJsonFromResource(String resourceName) throws IOException {
+    String jsonStr = IOUtils.toString(ClassLoader.getSystemResourceAsStream(resourceName), StandardCharsets.UTF_8);
+    return jsonStr.replaceAll("\\s+", "");
```
Why was this changed? Does Elasticsearch no longer accept whitespace in its json?

jywadhwani replied:
The JSON returned by `searchRequest.source()` was single-line with no whitespace. On the other hand, JSON read by the method `loadJsonFromResource` earlier was in multi-line form. Hence needed the conversion for asserts to work.

jplaisted replied:
Ah, ok, this isn't used for requests, but verifying responses, got it.

jywadhwani commented:
> I think this is fine for now.
> 
> Thinking about it, a "better" way, if we do something similar in the future, would be to have a PR with two commits. The first commit just verbatim copies the code. The second commit modifies it. Then I can view the diff between the two commits. Not worth messing around with for now, but we can consider that in the future.

Agree, worth considering in future.

jywadhwani commented on dao-impl/elasticsearch-dao-7/src/test/java/com/linkedin/metadata/dao/browse/BrowseDAOTest.java:
```diff
@@ -0,0 +1,103 @@
+        "/School/Characteristics/General/Embedding/network_standardized_school_embeddings_v3");
+  }
+
+
```
Need to enable test for this method, for some reason it got disabled. Working on it.

jplaisted commented:
fyi, with github, you don't need to do amend / force pushes. in fact, it's probably bad. I can't see what you've changed in that last push, unlike with review board.

When you merge the PR you have the opportunity to squash all the commits in the PR, so have many small commits with comments like "address comments" are totally fine since you'll wipe them out later.

tl;dr review board = one commit ever, github = each change is a new commit (please), clean up at submit

jywadhwani commented:
> fyi, with github, you don't need to do amend / force pushes. in fact, it's probably bad. I can't see what you've changed in that last push, unlike with review board.
> 
> When you merge the PR you have the opportunity to squash all the commits in the PR, so have many small commits with comments like "address comments" are totally fine since you'll wipe them out later.
> 
> tl;dr review board = one commit ever, github = each change is a new commit (please), clean up at submit
> 
> edit: for example, check out #12. Many commits in the PR, but only one final commit in the history when it was merged.

The last commit was related to `BrowseDAOTest`.
Regardless, I have separated changes to multiple commits to make the review easier. Hope this addresses your concerns.

jplaisted reviewed (approved):
LGTM, be sure to do "squash and merge", then clean up the description.
